### PR TITLE
Move `resource_usage` to roxy

### LINKED ISSUE #17: Move `resource_usage` to roxy
Since oinq is for communicating commands and roxy for doing them, getting resource usage would rather be moved to roxy.

### PR DESCRIPTION
**Pull request by sehkone**
This PR is dependent on [roxy's](https://github.com/aicers/roxy/pull/84). This is referring to the branch of the roxy's PR but will be changed to refer to the roxy main when it is merged into the main. This closes #17.

### CODE REVIEW

**Comment by sehkone** [comment]
@msk ,

When do users need to call `request::handle_resource_usage`?

**Comment by msk** [comment]
Pending until aicers/roxy#86 is resolved and all the other projects (piglet, reconverge, and hog) have been converted to use roxy.

**Comment by sehkone** [comment]
> Pending until [aicers/roxy#86](https://github.com/aicers/roxy/issues/86) is resolved and all the other projects (piglet, reconverge, and hog) have been converted to use roxy.

Since [aicers/roxy#86](https://github.com/aicers/roxy/issues/86) has been resolved and the other projects need this version of oinq to implement their using roxy, @msk, would you proceed this?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macOS-latest, windows-latest]
```
Comment (msk) [inline .github/workflows/ci.yml:28]: What build/test error do you get on Windows?

File: src/request.rs [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 
-use std::path::Path;
-
 use async_trait::async_trait;
 use bincode::Options;
-use gethostname::gethostname;
 use num_enum::FromPrimitive;
 use quinn::{RecvStream, SendStream};
-use serde::{Deserialize, Serialize};
+use roxy::ResourceUsage;
```
Comment (msk) [inline src/request.rs:7]: Since `ResouceUsage` is a part of the API, it should be defined in oinq.

File: Cargo.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 name = "oinq"
 version = "0.5.0"
 edition = "2021"
-rust-version = "1.64"
+rust-version = "1.65"
```
Comment (msk) [inline Cargo.toml:5]: Does oinq use anything unavailable in 1.64?

**Review by msk — CHANGES_REQUESTED** [review]

**Review by msk — APPROVED** [review]

**Pull request merged**